common/src/main/java/org/opensearch/ml/common/breaker/DiskCircuitBreaker.java

```diff
@@ -33,6 +32,12 @@ public DiskCircuitBreaker(long threshold, String diskDir) {
 
     @Override
     public boolean isOpen() {
-        return (new File(diskDir).getFreeSpace()/1024/1024/1024) < getThreshold();  // in GB
+        try {
+            return AccessController.doPrivileged((PrivilegedExceptionAction<Boolean>) () -> {
+                return (new File(diskDir).getFreeSpace()/1024/1024/1024) < getThreshold();  // in GB
+            });
+        } catch (PrivilegedActionException e) {
+            throw new MLException("Failed to run disk circuit breaker");
+        }
```

Comment:
https://github.com/opensearch-project/ml-commons/blob/2.x/plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java#L88. I think the access control added here can be removed. The try catch looks redundant here and needs unnecessary tests coverage.

Reply:
This is to avoid catching `PrivilegedActionException` everywhere using this method. We can save some effort by removing this try catch for this method, but will add more efforts for everywhere using this method. If more and more places using this method, we need to catch `PrivilegedActionException` for all of these places and that seems much more effort to do test coverage.

Reply:
We are talking about the same thing. Check the [PR](https://github.com/opensearch-project/ml-commons/pull/503). I meant to remove the try catch in the TaskRunner because the access control is added in the circuit breaker.

Reply:
Sorry, misunderstood. Good catch.